PDF_converter

krassovki opened the pull request:
- Added main
- Added tests

krassovki committed:
PDF_converter

- Added main
- Added tests

glock3 commented on Misha/Files/pdf_converter/pdf_converter_main.py:
```diff
@@ -0,0 +1,50 @@
+        f.close()
+        return text
+    except FileNotFoundError:
+        raise FileNotFoundError('No such file or directory')
```
Why do you need this error rise?

glock3 commented on Misha/Files/pdf_converter/pdf_converter_tests.py:
```diff
@@ -0,0 +1,31 @@
+from pdf_converter_main import read_from_file, change_format_to_pdf
+
+
+class PDFConverterTest(unittest.TestCase):
```
Please add more tests
1) When big single line converted to pdf
2) When second page is used

glock3 replied:
Also use separate folder (test-resource) for test files

glock3 commented on Misha/Files/pdf_converter/pdf_converter_main.py:
```diff
@@ -0,0 +1,50 @@
+    #test_file.html
+    #test_file.ibc
+    #test_file.txt
+    text = read_from_file(path)
```
File format is not only about last three letters . In your code you are always convert text file but with different extension

glock3 commented on Misha/Files/pdf_converter/pdf_converter_tests.py:
```diff
@@ -0,0 +1,31 @@
+
+    def test_read_from_file_with_wrong_path(self):
+        with self.assertRaises(FileNotFoundError) as context:
+            self.assertRaises(FileNotFoundError, read_from_file('path/to/file/'))
```
Do you need this exception check? Do you need double exception check?

glock3 commented on Misha/Files/pdf_converter/pdf_converter_main.py:
```diff
@@ -0,0 +1,50 @@
+        raise FileNotFoundError('No such file or directory')
+
+
+def change_format_to_pdf(path):
```
I don't think that you need to change file extension manually... You shouldn't do it at all

krassovki committed:
PDF-converter 2nd attempt

- Added test for long string
- Changed way to write info in pdf file

krassovki committed:
pdf_converter_3

- Deleted useless raise
- Deleted useless test
- Asserts in tests should be as they are

krassovki committed:
pdf_converter

- Corrected all asserts

glock3 commented on Misha/Files/pdf_converter/pdf_converter_tests.py:
```diff
@@ -0,0 +1,37 @@
+            actual_answer = True
+        else:
+            actual_answer = False
+        self.assertEqual(expected_answer, actual_answer)
```
use assertTrue

glock3 commented on Misha/Files/pdf_converter/pdf_converter_tests.py:
```diff
@@ -0,0 +1,37 @@
+        path_to_pdf = change_format_to_pdf('longstr.txt')
+        convert_to_pdf(text, path_to_pdf)
+        expected_answer = True
+        if os.path.exists(path_to_pdf):
```
Simplify this lines :)

glock3 commented on Misha/Files/pdf_converter/pdf_converter_tests.py:
```diff
@@ -0,0 +1,37 @@
+        self.assertEqual(expected_answer, actual_answer)
+
+
+if __name__ == '__main__':
```
Remove this main from test

krassovki committed:
pdf_converter

- AssertEqual() changed to AssertTrue
